spec.md

```diff
@@ -137,11 +137,15 @@ Typically, the first step in pulling an artifact is to retrieve the manifest. Ho
 To pull a manifest, perform a `GET` request to a URL in the following form:
 `/v2/<name>/manifests/<reference>` <sup>[end-3](#endpoints)</sup>
 
-`<name>` refers to the namespace of the repository. `<reference>` MUST be either (a) the digest of the manifest or (b) a tag name.
+`<name>` refers to the namespace of the repository. `<reference>` MUST be either (a) the digest of the manifest or (b) a tag.
 The `<reference>` MUST NOT be in any other format. Throughout this document, `<name>` MUST match the following regular expression:
 
 `[a-z0-9]+([._-][a-z0-9]+)*(/[a-z0-9]+([._-][a-z0-9]+)*)*`
 
+Throughout this document, `<reference>` as a tag MUST be at most 128 characters in length and MUST match the following regular expression:
+
+`[a-zA-Z0-9_]+[a-zA-Z0-9._-]*`
```

Comment:
```suggestion
`[a-zA-Z0-9_][a-zA-Z0-9._-]{0,127}`
```

Is there a reason for not adding the quantifier?

For reference, `[\w][\w.-]{0,127}` is what we're using in [distribution/distribution](https://github.com/distribution/distribution). I'm fine with not using `\w`, but I'd keep the length.